src/worker/vm/extensions/helpers/api-key-manager.ts

```diff
@@ -38,19 +38,22 @@ export class PluginsApiKeyManager {
         try {
             let key: string | null = null
             const userResult = await this.db.postgresQuery(
-                `SELECT id FROM posthog_user WHERE email = $1`,
-                [PLUGINS_API_KEY_USER_EMAIL],
+                `SELECT id FROM posthog_user WHERE email LIKE $1`,
+                [`%@${POSTHOG_BOT_USER_EMAIL_DOMAIN}`],
                 'fetchPluginsUser'
             )
 
             if (userResult.rowCount < 1) {
+                const botUserEmailId = Math.round(Math.random() * 100000000)
```

Comment:
Why not `organizationId`?

Reply:
I was thinking this is useful if we do later decide to add more bot users

Reply:
Most proper-domain-like-but-still-dummy would be something like below
```TypeScript
`${organizationId}@system.posthog`
```
 (and I _hope_ we don't add more bot users because that'd be too much 😅), but current scheme should be alright too.

Reply:
Well for example [this comment](https://github.com/PostHog/plugin-server/pull/519#pullrequestreview-730393631) brings up one approach that would lead to multiple bot users